core/src/main/java/org/apache/sedona/core/spatialRDD/SpatialRDD.java

```diff
@@ -564,7 +588,45 @@ public void saveAsWKT(String outputLocation)
             }
         }).saveAsTextFile(outputLocation);
     }
-
+    
+    public void saveAsParquet(JavaSparkContext sc,
+                              String geometryColumn,
+                              List<Field> userColumns,
+                              String outputLocation,
+                              String namespace,
+                              String name) throws SedonaException, IOException {
+        sc.hadoopConfiguration().setBoolean("parquet.enable.summary-metadata", false);
+        final Job job = Job.getInstance(sc.hadoopConfiguration());
+        GeometryType geometryType = this.getGeometryType();
+        Schema schema = AvroUtils.getSchema(geometryType, geometryColumn, userColumns,namespace,
+                                            name);
+//        ParquetOutputFormat.setWriteSupportClass(job, AvroWriteSupport.class);
+        AvroParquetOutputFormat.setSchema(job, schema);
+        this.rawSpatialRDD.mapPartitionsToPair(new PairFlatMapFunction<Iterator<T>, Void, GenericRecord>() {
+            @Override
+            public Iterator<Tuple2<Void, GenericRecord>> call(Iterator<T> iterator) throws Exception {
+                Schema schema = AvroUtils.getSchema(geometryType, geometryColumn, userColumns, namespace,
+                                                    name);
+                Iterable<T> recordIterable = () -> iterator;
+                try {
+                    return StreamSupport.stream(recordIterable.spliterator(), false).map(geometry -> {
+                        try {
+                            GenericRecord genericRecord =
+                                    AvroUtils.getRecord(geometry, geometryType, geometryColumn, schema);
```

Comment:
Why are we not simply choosing WKB for the serialization of the geometry column and keep the rest as a normal parquet file?